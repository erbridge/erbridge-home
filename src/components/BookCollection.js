import PropTypes from 'prop-types';
import React, { Component } from 'react';

import OpenLibrary from '../helpers/OpenLibrary';

import Book from './Book';

class BookCollection extends Component {
  static propTypes = {
    isbns: PropTypes.arrayOf(PropTypes.string).isRequired,
  };

  state = {
    books: [],
  };

  async fetchBooks() {
    const { isbns } = this.props;

    this.setState({ books: [] });
    this.setState({ books: await OpenLibrary.fetchBooks(isbns) });
  }

  async componentDidMount() {
    await this.fetchBooks();
  }

  async componentDidUdpate(prevProps) {
    const oldIsbns = prevProps.isbns;
    const newIsbns = this.props.isbns;

    if (
      newIsbns.length !== oldIsbns.length ||
      newIsbns.some((isbn, i) => isbn !== oldIsbns[i])
    ) {
      await this.fetchBooks();
    }
  }

  render() {
    const { books } = this.state;

    return (
      <div className="BookCollection">
        {books.map(book => (
          <Book key={book.identifiers.openlibrary} {...book} />
        ))}
      </div>
    );
  }
}

export default BookCollection;
