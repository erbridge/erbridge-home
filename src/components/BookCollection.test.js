import { shallow } from 'enzyme';
import React from 'react';

import BookCollection from './BookCollection';

jest.mock('../helpers/OpenLibrary');

it('renders correctly with all props', async () => {
  const wrapper = shallow(
    <BookCollection isbns={['0123456789', '0123456789ABC']} />,
  );

  await waitForNextTick();

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly with empty isbns', async () => {
  const wrapper = shallow(<BookCollection isbns={[]} />);

  await waitForNextTick();

  expect(wrapper).toMatchSnapshot();
});
