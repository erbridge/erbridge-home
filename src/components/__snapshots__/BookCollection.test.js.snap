// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<div
  className="BookCollection"
>
  <Book
    authors={
      Array [
        Object {
          "name": "Author A for 0123456789",
        },
        Object {
          "name": "Author B for 0123456789",
        },
      ]
    }
    identifiers={
      Object {
        "isbn_10": "0123456789",
        "isbn_13": undefined,
        "openlibrary": "OL0123456789",
      }
    }
    key="OL0123456789"
    subtitle="Subtitle for 0123456789"
    title="Title for 0123456789"
    url="http://example.com/0123456789"
  />
  <Book
    authors={
      Array [
        Object {
          "name": "Author A for 0123456789ABC",
        },
        Object {
          "name": "Author B for 0123456789ABC",
        },
      ]
    }
    identifiers={
      Object {
        "isbn_10": undefined,
        "isbn_13": "0123456789ABC",
        "openlibrary": "OL0123456789ABC",
      }
    }
    key="OL0123456789ABC"
    subtitle="Subtitle for 0123456789ABC"
    title="Title for 0123456789ABC"
    url="http://example.com/0123456789ABC"
  />
</div>
`;

exports[`renders correctly with empty isbns 1`] = `
<div
  className="BookCollection"
/>
`;
